src/extension.ts

```diff
@@ -2,44 +2,49 @@ import * as vscode from "vscode";
 import * as jsxInfo from "jsx-info";
 import * as logger from "./logger";
 
-export function activate(_context: vscode.ExtensionContext) {
+export function activate(context: vscode.ExtensionContext) {
   const jsxInfoProvider = new JSXInfoProvider();
   const treeView = vscode.window.createTreeView("jsxInfo", {
     treeDataProvider: jsxInfoProvider,
     showCollapseAll: true,
   });
-  vscode.commands.registerCommand("jsxInfo.run", async () => {
+  async function cmdRun() {
     try {
       await jsxInfoProvider.run(treeView);
     } catch (err) {
       vscode.window.showErrorMessage(err.message);
       logger.fail(err.message);
     }
-  });
-  vscode.commands.registerCommand("jsxInfo.refresh", async () => {
+  }
+  async function cmdRefresh() {
     try {
       await jsxInfoProvider.refresh(treeView);
     } catch (err) {
       vscode.window.showErrorMessage(err.message);
       logger.fail(err.message);
     }
-  });
-  vscode.commands.registerCommand(
-    "jsxInfo._openFile",
-    async (options: {
-      filename: string;
-      startLine: number;
-      startColumn: number;
-      endLine: number;
-      endColumn: number;
-    }) => {
-      try {
-        await openFile(options);
-      } catch (err) {
-        vscode.window.showErrorMessage(err.message);
-        logger.fail(err.message);
-      }
+  }
+  async function cmdOpenFile(options: {
+    filename: string;
+    startLine: number;
+    startColumn: number;
+    endLine: number;
+    endColumn: number;
+  }) {
+    try {
+      await openFile(options);
+    } catch (err) {
+      vscode.window.showErrorMessage(err.message);
+      logger.fail(err.message);
     }
+  }
+  // Dispose the commands when the extension is deactivated
+  context.subscriptions.push(
+    vscode.commands.registerCommand("jsxInfo.run", cmdRun),
+    vscode.commands.registerCommand("jsxInfo.refresh", cmdRefresh),
+    // This command isn't exported in package.json because it's an internal
+    // thing used be the tree view
+    vscode.commands.registerCommand("jsxInfo._openFile", cmdOpenFile)
```

Comment:
This is proper cleanup for folks who want to disable the extension without restarting vscode